packages/notebook/src/browser/service/notebook-options.ts

```diff
@@ -0,0 +1,123 @@
+    NotebookPreferences.OUTPUT_FONT_SIZE,
+    NotebookPreferences.OUTPUT_FONT_FAMILY,
+    NotebookPreferences.OUTPUT_SCROLLING,
+    NotebookPreferences.OUTPUT_WORD_WRAP,
+    NotebookPreferences.OUTPUT_LINE_LIMIT
+];
+
+export interface NotebookOutputOptions {
+    // readonly outputNodePadding: number;
+    // readonly outputNodeLeftPadding: number;
+    // readonly previewNodePadding: number;
+    // readonly markdownLeftMargin: number;
+    // readonly leftMargin: number;
+    // readonly rightMargin: number;
+    // readonly runGutter: number;
+    // readonly dragAndDropEnabled: boolean;
+    readonly fontSize: number;
+    readonly outputFontSize?: number;
+    readonly fontFamily: string;
+    readonly outputFontFamily?: string;
+    // readonly markupFontSize: number;
+    // readonly markdownLineHeight: number;
+    readonly outputLineHeight: number;
+    readonly outputScrolling: boolean;
+    readonly outputWordWrap: boolean;
+    readonly outputLineLimit: number;
+    // readonly outputLinkifyFilePaths: boolean;
+    // readonly minimalError: boolean;
+
+}
+
+@injectable()
+export class NotebookOptionsService {
+
+    @inject(PreferenceService)
+    protected readonly preferenceService: PreferenceService;
+
+    @inject(EditorPreferences)
+    protected readonly editorPreferences: EditorPreferences;
+
+    protected outputOptionsChangedEmitter = new Emitter<NotebookOutputOptions>();
+    onDidChangeOutputOptions = this.outputOptionsChangedEmitter.event;
+
+    @postConstruct()
+    init(): void {
+        this.preferenceService.onPreferencesChanged(async preferenceChanges => {
+            if (notebookOutputOptionsRelevantPreferences.some(p => p in preferenceChanges)) {
+                this.outputOptionsChangedEmitter.fire(this.computeOutputOptions());
+            }
+        });
+    }
+
+    computeOutputOptions(): NotebookOutputOptions {
+        const outputLineHeight = this.preferenceService.get<number>(NotebookPreferences.OUTPUT_LINE_HEIGHT)!;
+
+        const fontSize = this.preferenceService.get<number>('editor.fontSize')!;
+        const outputFontSize = this.preferenceService.get<number>(NotebookPreferences.OUTPUT_FONT_SIZE)!;
+
+        return {
+            fontSize,
+            outputFontSize: this.preferenceService.get<number>(NotebookPreferences.OUTPUT_FONT_SIZE),
+            fontFamily: this.preferenceService.get<string>('editor.fontFamily')!,
+            outputFontFamily: this.preferenceService.get<string>(NotebookPreferences.OUTPUT_FONT_FAMILY),
+            outputLineHeight: this.computeOutputLineHeight(outputLineHeight, outputFontSize ?? fontSize),
+            outputScrolling: this.preferenceService.get<boolean>(NotebookPreferences.OUTPUT_SCROLLING)!,
+            outputWordWrap: this.preferenceService.get<boolean>(NotebookPreferences.OUTPUT_WORD_WRAP)!,
+            outputLineLimit: this.preferenceService.get<number>(NotebookPreferences.OUTPUT_LINE_LIMIT)!
+        };
```

Comment:
Suggestion: Using the nullable coercion (`!`) operator for the `preferenceService#get` call is not ideal. Use the second parameter of `get` to specify the default value in case the preference value is invalid.

Reply:
But these specific settings with nullable coercion allready have default values. So they should be always set. Should i rather write this as `this.preferenceService.get<boolean>(NotebookPreferences.OUTPUT_WORD_WRAP, notebookPreferenceSchema.properties[NotebookPreferences.OUTPUT_FONT_FAMILY].default as string)`?

Reply:
One could argue that adopters could remove some of these preferences, thereby deleting the default value. I find this scenario to be highly unlikely though. I think we can keep it as it is for now.

Reply:
ok should be a little better now

Reply:
Yes, perfect, thanks!